packages/protocol/contracts/Staking.sol

```diff
@@ -130,6 +132,7 @@ contract Staking is ERC900, IStakingLocking, IsContract, TimeHelpers {
      * @param _amount Number of tokens to be transferred
      */
     function transferAndUnstake(address _to, uint256 _amount) external {
+        // Note: I understand this is to re-use functions, but it would be a nice to skip the extra transfer in-between and just inline a function to emit an event
```

Comment:
This is the only logic change I would push for, since the extra storage costs for the two separate actions can get quite high (multiple SLOADs and SSTOREs).